File: fold-an-array/fold-an-array.js
function foldArray(arr, runs){

for(var i=0;i<runs;i++){arr = fold(arr)}
return arr
}

function fold(ar){
var l = ar.length;
var t = [];
if (l%2==1){ for(j=0;j<Math.floor(l/2);j++){t.push(ar[j] + ar[l-j-1])}
t.push(ar[Math.floor(l/2)]);
return t;
}
else{for(j=0;j<l/2;j++){t.push(ar[j] + ar[l-j-1])}return t}
}

File: fold-an-array/fold-an-array.py
def fold_array(arr, runs):
    i = 0
    while i < runs:
        arr = fold(arr)
        i+=1
    return arr
        
def fold(ar):
    l = len(ar)
    if l % 2 == 1:
        return [ar[i] + ar[l-i-1] for i in range(l//2)] + [ar[l//2]]
    else:
        return [ ar[i] + ar[l-i-1] for i in range(l//2)]